src/NHibernate/Linq/InsertSyntax.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using Remotion.Linq;
+using Remotion.Linq.Utilities;
+
+namespace NHibernate.Linq
+{
+	public class InsertSyntax<TInput>
+	{
+		private readonly IQueryable<TInput> _query;
+
+		internal InsertSyntax(IQueryable<TInput> query)
+		{
+			_query = query;
+		}
+
+		/// <summary>
+		/// Executes the insert, using the specified assignments.
+		/// </summary>
+		/// <typeparam name="TOutput">The type of the output.</typeparam>
+		/// <param name="assignmentActions">The assignments.</param>
+		/// <returns></returns>
+		public int Into<TOutput>(Action<Assignments<TInput, TOutput>> assignmentActions)
```

Comment:
Maybe renaming `TInput` and `TOutput` as `TSource` and `TTarget` would be more explicit.
And comment doc for TTarget could be "The type of the entities to insert."